### PR TITLE
boot: Load Options

### PR DESCRIPTION
**Pull request by hugelgupf**
This will help with testing. Introduces a new option to do everything but actually kexec_load.

I've always felt luke-warm on this API style, but used it sometimes.

In this case, I didn't want an API that read like `Load(false, true)` when calling it if we made it `Load(verbose, dryRun bool)` or something like that.

### CODE REVIEW

**Comment by 10000TB** [comment]
how will this be used ?

File: pkg/boot/boot.go [hunk 1]
```diff
@@ -10,8 +10,42 @@ import (
 	"fmt"
 
 	"github.com/u-root/u-root/pkg/boot/kexec"
+	"github.com/u-root/uio/ulog"
 )
 
+// LoadOption is an optional parameter to Load.
+type LoadOption func(o *loadOptions)
```
Comment (rminnich) [inline pkg/boot/boot.go:17]: I forget, but can't this just be
type LoadOption func(*loadOptions)

**Review by rminnich — COMMENTED** [review]
This seems kinda neat. I'm not sure what it means if options in the ...loadoptions conflict? What is one sets the kexec option one way and another sets it another way? Will this surprise people? Still thinking.

**Comment by hugelgupf** [comment]
> This seems kinda neat. I'm not sure what it means if options in the ...loadoptions conflict? What is one sets the kexec option one way and another sets it another way? Will this surprise people? Still thinking.

Sort of typical for this kind of API. The later specification wins. In this case, however, I made it such that you can't specify conflicting options, because conflicting ones don't exist: there's no `WithNoDryRun` option to negate `WithDryRun`, and there's no `WithNoVerbose`. You *can* do `Load(WithLogger(log.Default()), WithLogger(ulog.Null))`, though, and the latter will be used.

**Review by rminnich — DISMISSED** [review]

**Pull request merged**